### ISSUE TITLE
How do I use subtraction results of two columns in an array order statement

### ISSUE THREAD

**Issue by rbjarnason** [sequelize/sequelize#4558]
This works:
```
findAll({
  ...
  order: "(counter_endorsements_up-counter_endorsements_down) DESC"
  ...
});
```

This does not work:
```
order: [
   "(counter_endorsements_up-counter_endorsements_down) DESC",
   [ { model: models.Image, as: 'IdeaHeaderImages' } ,'updated_at', 'asc' ]
 ]
```

I tried this, but no such function available:
```
order: [
   [models.sequelize.fn('subtraction', models.sequelize.col('counter_endorsements_up'), models.sequelize.col('counter_endorsements_down')), 'DESC'],
   [ { model: models.Image, as: 'IdeaHeaderImages' } ,'updated_at', 'asc' ]
 ]
```

Any ideas?

**Comment by janmeier** [comment]
Try wrapping the substraction in `sequelize.literal`

**Comment by rbjarnason** [comment]
Thanks @janmeier it works :)

**Comment by rbjarnason** [comment]
@janmeier The sequelize.literal approach stops working if I also use limit and offset for pagination.

What is the correct approach to use subtraction?

I've tried:
[models.sequelize.fn('subtract', models.sequelize.col('counter_endorsements_up'), models.sequelize.col('counter_endorsements_down')), 'DESC']

and:
[models.sequelize.fn('-', models.sequelize.col('counter_endorsements_up'), models.sequelize.col('counter_endorsements_down')), 'DESC']

And get  no such function available.

Just to test I used power and that worked but does not give me the right results:
[models.sequelize.fn('pwer', models.sequelize.col('counter_endorsements_up'), models.sequelize.col('counter_endorsements_down')), 'DESC']

Any hints?
